sabnzbd/decoder.py

```diff
@@ -172,6 +172,14 @@ def run(self):
                     elif not killed and not found:
                         logme = T('Badly formed yEnc article in %s') % art_id
                         logging.info(logme)
+                        # bad yEnc article, so ... let's check if it's uuencode, to inform user and stop download
+                        for line in data_to_check:
```

Comment:
the only line this code uses for detecting uuencoding is always at the start of the article body. Do you really want to loop over the entire article here?

Reply:
I thought so, but it is not; the first line is a header part, then one or two newlines, then \nbegin.

I did some debugging: the find was often in position 36.

And I do a break after finding it.

Reply:
Hence why I wrote 'start of the article body' - i.e. the part directly following the headers. Thing is, if the problem isn't uuencoding the loop continues over all data of every affected article, for every configured server.

Even imposing a very basic limit on the number of lines to check (like the [old code](https://github.com/sabnzbd/sabnzbd/blob/2.3.x/sabnzbd/decoder.py#L249)) would avoid a lot of futile work. Or maybe check the ending instead, that ```\n`\nend\n.``` stuff?

Reply:
OK. I can add a maxline-checker. Or: How about only inspecting the first 1000 bytes?

Reply:
@jcfp/@sanderjo in the old code the data was already split in lines, but here we still have raw data from the socket. So here the `data_to_check` will be much smaller. I'd recommend maybe only checking the first 2 or 3.

This reminds me that I need to remove the `nw.lines` stuff, since from Py3 we only work with the raw data.

Reply:
I updated `develop`, have to use `raw_data` now instead of `data_to_check`.